### PR TITLE
Convert rz_print_date_* to rz_time_date_*

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Cleaning up time.c code to remove rz_print code.

### CODE REVIEW

File: librz/util/time.c [hunk 1]
```diff
@@ -135,96 +162,55 @@ RZ_API bool rz_time_stamp_is_dos_format(const ut32 certainPosixTimeStamp, const
...
+	time_t t = (time_t)timestamp; // TODO: limit above!
+	return rz_time_stamp_to_str(t);
+}
+
+/**
+ * \brief Returns the timestamp in string format of the current time (now)
+ *
+ * \return The timestamp in string format
+ */
+RZ_API RZ_OWN char *rz_time_date_now_to_string() {
```
Comment (ret2libc) [inline librz/util/time.c]:
```suggestion
RZ_API RZ_OWN char *rz_time_date_now_to_string(void) {
```
?

File: librz/util/time.c [hunk 2]
```diff
@@ -135,96 +162,55 @@ RZ_API bool rz_time_stamp_is_dos_format(const ut32 certainPosixTimeStamp, const
...
+
+/**
+ * \brief Converts a Win32 date (ut64) and returns the timestamp in string format
+ *
+ * \param  timestamp The number to convert to string
+ * \return           The timestamp in string format
+ */
+RZ_API RZ_OWN char *rz_time_date_w32_to_string(ut64 timestamp) {
+	ut64 limit = 0x2b6109100ll; // seconds to start of 1970
+	timestamp /= 10000000ll; // 100 nanoseconds to seconds
```
Comment (ret2libc) [inline librz/util/time.c]: Where do these come from? Maybe use a define to avoid magic numbers

File: librz/util/time.c [hunk 3]
```diff
@@ -135,96 +162,55 @@ RZ_API bool rz_time_stamp_is_dos_format(const ut32 certainPosixTimeStamp, const
...
+}
+
+/**
+ * \brief Converts a Mac HFS+ date (ut32) and returns the timestamp in string format
+ *
+ * \param  timestamp The number to convert to string
+ * \return           The timestamp in string format
+ */
+RZ_API RZ_OWN char *rz_time_date_hfs_to_string(ut32 timestamp) {
+	timestamp += 2082844800; // add Mac HFS+ epoch
```
Comment (ret2libc) [inline librz/util/time.c]: magic number

File: librz/core/cmd/cmd_print.c [hunk 4]
```diff
@@ -1584,87 +1584,90 @@ static void cmd_print_gadget(RzCore *core, const char *_input) {
...
-	int len = core->blocksize;
+	char *date = NULL;
+	const ut8 *block = core->block;
+	ut32 len = core->blocksize;
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
 	if (len < sizeof(ut64)) {
-		eprintf("Please change the block size to a value greater than and a multiple of %zu. For example, "
-			"run `b %zu`\n",
-			sizeof(ut64), sizeof(ut64));
+		RZ_LOG_ERROR("The block size less than 8.\n");
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
		RZ_LOG_ERROR("The block size is less than 8.\n");
```

File: librz/core/cmd/cmd_print.c [hunk 5]
```diff
@@ -1584,87 +1584,90 @@ static void cmd_print_gadget(RzCore *core, const char *_input) {
...
+	char *date = NULL;
+	const ut8 *block = core->block;
+	ut32 len = core->blocksize;
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+	int timezone = (int)rz_config_get_i(core->config, "time.zone");
 	if (len < sizeof(ut32)) {
-		eprintf("Please change the block size to a value greater than and a multiple of %zu. For example, "
-			"run `b %zu`\n",
-			sizeof(ut32), sizeof(ut32));
+		RZ_LOG_ERROR("The block size less than 4.\n");
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
		RZ_LOG_ERROR("The block size is less than 4.\n");
```

File: librz/core/cmd/cmd_print.c [hunk 6]
```diff
@@ -1584,87 +1584,90 @@ static void cmd_print_gadget(RzCore *core, const char *_input) {
...
-	int len = core->blocksize;
+	char *date = NULL;
+	const ut8 *buf = NULL;
+	const ut8 *block = core->block;
+	ut32 len = core->blocksize;
 	if (len < sizeof(ut32)) {
-		eprintf("Please change the block size to a value greater than and a multiple of %zu. For example, "
-			"run `b %zu`\n",
-			sizeof(ut32), sizeof(ut32));
+		RZ_LOG_ERROR("The block size less than 4.\n");
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
		RZ_LOG_ERROR("The block size is less than 4.\n");
```

File: librz/core/cmd/cmd_print.c [hunk 7]
```diff
@@ -1584,87 +1584,90 @@ static void cmd_print_gadget(RzCore *core, const char *_input) {
...
+	char *date = NULL;
+	const ut8 *block = core->block;
+	ut32 len = core->blocksize;
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+	int timezone = (int)rz_config_get_i(core->config, "time.zone");
 	if (len < sizeof(ut32)) {
-		eprintf("Please change the block size to a value greater than and a multiple of %zu. For example, "
-			"run `b %zu`\n",
-			sizeof(ut32), sizeof(ut32));
+		RZ_LOG_ERROR("The block size less than 4.\n");
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
		RZ_LOG_ERROR("The block size is less than 4.\n");
```

**Review by ret2libc — COMMENTED** [review]
Much better this way!

File: librz/util/time.c (hunk 2 above)
Comment (wargio) [reply librz/util/time.c]: these comes from the windows stuff. only used there :/

File: librz/core/cmd/cmd_print.c [hunk 8]
```diff
@@ -1584,87 +1584,90 @@ static void cmd_print_gadget(RzCore *core, const char *_input) {
...
 		return RZ_CMD_STATUS_ERROR;
 	}
-	if (len % sizeof(ut32)) {
-		len = len - (len % sizeof(ut32));
-	}
-	for (l = 0; l < len; l += sizeof(ut32)) {
-		rz_print_date_dos(core->print, block + l, sizeof(ut32));
+	for (ut64 i = 0; i < len; i += sizeof(ut32)) {
+		buf = block + i;
+		ut32 dt = buf[3] << 24 | buf[2] << 16 | buf[1] << 8 | buf[0];
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: why not use an endian function?

**Review by XVilka — APPROVED** [review]
Overall is good.

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**